### PR TITLE
PTX-4560 Fix Upgrade hops test

### PR DESCRIPTION
**Pull request by nikolaypopov**
Pass MANIFEST_URL env var in case of edge-install and add check for empty expected PX version

### CODE REVIEW

File: pkg/util/test/util.go [hunk 1]
```diff
@@ -428,15 +428,16 @@ func validateStorageNodes(pxImageList map[string]string, cluster *corev1.Storage
 		if len(cluster.Spec.Env) > 0 {
 			for _, env := range cluster.Spec.Env {
 				if env.Name == PxReleaseManifestURLEnvVarName {
-					pxVersion = strings.TrimSpace(regexp.MustCompile(`\S+\/(\S+)\/version`).FindStringSubmatch(env.Value)[1])
-					if pxVersion == "" {
-						return fmt.Errorf("failed to extract version from value of %s", PxReleaseManifestURLEnvVarName)
-					}
+					expectedPxVersion = strings.TrimSpace(regexp.MustCompile(`\S+\/(\S+)\/version`).FindStringSubmatch(env.Value)[1])
```
Comment (piyush-nimbalkar) [inline pkg/util/test/util.go:431]: How do we handle master where the url does not have the version?

Comment (nikolaypopov) [reply pkg/util/test/util.go]: We haven't before, added support for handling master builds

File: pkg/util/test/util.go [hunk 2]
```diff
@@ -428,15 +431,22 @@ func validateStorageNodes(pxImageList map[string]string, cluster *corev1.Storage
 		if len(cluster.Spec.Env) > 0 {
 			for _, env := range cluster.Spec.Env {
 				if env.Name == PxReleaseManifestURLEnvVarName {
-					pxVersion = strings.TrimSpace(regexp.MustCompile(`\S+\/(\S+)\/version`).FindStringSubmatch(env.Value)[1])
-					if pxVersion == "" {
-						return fmt.Errorf("failed to extract version from value of %s", PxReleaseManifestURLEnvVarName)
+					// Looking for clear PX version before /version in the URL
+					ver := regexp.MustCompile(`\S+\/(\d.\S+)\/version`).FindStringSubmatch(env.Value)
+					if ver != nil {
+						expectedPxVersion = ver[1]
```
Comment (pure-jliao) [inline pkg/util/test/util.go:437]: `break` here if found?

File: pkg/util/test/util.go [hunk 3]
```diff
@@ -428,15 +431,22 @@ func validateStorageNodes(pxImageList map[string]string, cluster *corev1.Storage
...
+					} else { // If the above regex found nothing, assuming it was a master version URL
+						expectedPxVersion = PxMasterVersion
```
Comment (pure-jliao) [inline pkg/util/test/util.go:439]: we should set the master version outside of this env iteration loop if we didn't find anything matches

File: test/integration_test/basic_test.go [hunk 4]
```diff
@@ -187,6 +187,9 @@ func BasicUpgrade(tc *types.TestCase) func(*testing.T) {
 				// Set defaults
 				portworx.SetPortworxDefaults(cluster)
 
+				err = ci_utils.ConstructStorageCluster(cluster, hopURL, specImages)
```
Comment (pure-jliao) [inline test/integration_test/basic_test.go:190]: are we calling this to update the env variables? then we could replace `cluster.Spec.Image = specImages["version"]` with `err = ci_utils.ConstructStorageCluster(cluster, hopURL, specImages)`, it's safer to call `SetPortworxDefaults` afterwards.

File: pkg/util/test/util.go (hunk 2 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:437]: added break

File: pkg/util/test/util.go [hunk 5]
```diff
@@ -425,18 +428,26 @@ func validateStorageNodes(pxImageList map[string]string, cluster *corev1.Storage
...
 			for _, env := range cluster.Spec.Env {
 				if env.Name == PxReleaseManifestURLEnvVarName {
-					pxVersion = strings.TrimSpace(regexp.MustCompile(`\S+\/(\S+)\/version`).FindStringSubmatch(env.Value)[1])
-					if pxVersion == "" {
-						return fmt.Errorf("failed to extract version from value of %s", PxReleaseManifestURLEnvVarName)
+					// Looking for clear PX version before /version in the URL
+					ver := regexp.MustCompile(`\S+\/(\d.\S+)\/version`).FindStringSubmatch(env.Value)
+					if ver != nil {
+						expectedPxVersion = ver[1]
+						break
```
Comment (pure-jliao) [inline pkg/util/test/util.go:438]: nit: better to move break under `expectedPxVersion = PxMasterVersion`

**Review by pure-jliao — APPROVED** [review]
LGTM as long as it's tested. one code suggestion left.

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**